cdi_test/tests/run-cdi.sh

```diff
@@ -34,8 +34,9 @@ cdi_on_exit() {
 cdi_execute_cmd() {
     ip=$1
     cmd=$2
+    timeout=$3
     ssh -T -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null \
-        -o BatchMode=yes -o TCPKeepAlive=yes \
+        -o BatchMode=yes -o ${timeout} \
```

Comment:
I do not think ConnectTimeout sets the timeout for running the program via ssh. It only sets the timeout for establishing ssh connection

>ConnectTimeout Specifies the timeout (in seconds) used when connecting to the SSH server, instead of using the default system TCP timeout. This timeout is applied both to establishing the connection and to performing the initial SSH protocol handshake and key exchange.

Reply:
Does CDI test has timeout option or environment variables for timeout, like `MPIEXEC_TIMEOUT` for MPI ?

Reply:
It supposedly does, and it has a default value that's relatively small. I think I need to think of some way to timeout ssh running the program